feat: create user apps in uli community

maanasb01 opened the pull request:
This PR introduces a feature of creating user apps in the ULI community. 

Changes:

- New Schemas: `UserApplication` and `AppAccessToken`.
- I have kept the `UserApplication` table primary keys, UUIDs. The integer keys did not look natural while accessing the app page, for example: `/app/my-apps/4`. 
- The user can create new apps in the `/app/create` route. 
- User can see their apps in the `/app/my-apps' route.
- User can access the App page in the `/app/my-apps/{APP_ID}` route. 
  - On this page, the user can edit app details and also have an option to delete the app.
  - On this page, users can generate new app_access_tokens with a token name and expiry date. (The schema and table also have a field of access_level in case it is useful in the future. But in the form, I am not rendering that field). 
  - After token creation, a modal will be displayed with the generated token that can be copied. This is a one-time visible token and cannot be viewed again. 
  - On this page, users can see all the app tokens generated by them in a table. 
- Created a new plug and pipeline to authenticate app_access_tokens. 

Screenshots:
- Create App
<img width="3751" height="1944" alt="image" src="https://github.com/user-attachments/assets/83c3fff9-5e76-4273-a1b2-d766a18f02ef" />
- My-Apps Page
<img width="3751" height="1944" alt="image" src="https://github.com/user-attachments/assets/8f094101-9a15-4161-8375-87897c7f315c" />
- App page
<img width="3753" height="1944" alt="image" src="https://github.com/user-attachments/assets/81f658e5-134a-4d7e-94a1-2da45958753b" />
- Deleting App Warning
<img width="3753" height="1944" alt="image" src="https://github.com/user-attachments/assets/a30b51be-a201-4daf-8cf2-645437fb18ca" />
- Token Generation
<img width="3753" height="1944" alt="image" src="https://github.com/user-attachments/assets/4ee7fde9-4e58-4956-8597-6bd907dae5a3" />


TODO:
- Improve the UI.

dennyabrain commented on uli-community/lib/uli_community/api/app_access_token.ex:
```diff
@@ -0,0 +1,21 @@
+  schema "app_access_tokens" do
+    field :token_id, Ecto.UUID
+    field :token_name, :string
+    field :access_level, :string
```
Lets make this terminology consistent with the oAuth language. Lets call it `scope`
Example of oAuth scopes for mastodon can be seen here for your review `https://docs.joinmastodon.org/api/oauth-scopes/#granular`

Instead of string, this can be an array of strings. Because a token might have multiple scopes.

In our current iteration we can start with just two scopes - `dataset` and `classify`. We don't have to create a UI to select scopes for now. Every token created will have both these scopes for now. 
Note : don't make these the default value at the ecto schema level. Set them by default at the controller level whenever a new token is created.

dennyabrain commented:
Some general UI feedback : 
1. Not related to this PR itself but lets change the background of the website back to white. The light orange brand color is not suited for being used as a background color. Its also making other white background components look weird.
2. For pages without a lot of content, like the "Create a new User App" page, constrain the page width to be narrow, so that the UI within does not fill up the entire width. This narrow column can then be left aligned
3. In the app details page, see if the "test App" thing can be made into a card (a fixed width component with a light border). 
4. All the popups are also unusually wide. See if they can be constrained and only take full width on mobile devices.

maanasb01 commented:
Hey @dennyabrain , I have pushed the new changes:
- Changed access_level to "scope", and generating them by default from the controller only. 
- Fix the modal width.
- Converted the App details to a card, also modified the "back to my apps" button to look better. 
- Shorten the width of pages that don't have much content, like: create app. 
- For other UI fixes, I am planning to fix them in a separate PR.

Screenshots:
<img width="3750" height="1945" alt="image" src="https://github.com/user-attachments/assets/3980d710-7621-4470-b95b-d45f7ed56438" />
 
<img width="3750" height="1945" alt="image" src="https://github.com/user-attachments/assets/6e1738ad-8c2e-4aa0-828d-43388b52d4fd" />
<img width="3750" height="1945" alt="image" src="https://github.com/user-attachments/assets/d5d1c74e-674b-464c-8058-cc512a2d455c" />
